core/state_transition.go

```diff
@@ -269,6 +269,13 @@ func (st *stateTransition) buyGas() error {
 			mgval = mgval.Add(mgval, l1Cost)
 		}
 	}
+	var operatorCost *big.Int
+	if st.evm.Context.OperatorCostFunc != nil && !st.msg.SkipNonceChecks && !st.msg.SkipFromEOACheck {
+		operatorCost = st.evm.Context.OperatorCostFunc(new(big.Int).SetUint64(st.msg.GasLimit), false, st.evm.Context.Time)
+		if operatorCost != nil {
+			mgval = mgval.Add(mgval, operatorCost)
+		}
+	}
 	balanceCheck := new(big.Int).Set(mgval)
 	if st.msg.GasFeeCap != nil {
 		balanceCheck.SetUint64(st.msg.GasLimit)
```

Comment:
It seems like a bug that `lCost` is being added in this branch. agree that operator cost should not added in this branch. we should not change anything in this PR related to this comment

Reply:
Wow.. good catch.

Reply:
Want me to open a PR to fix?

Reply:
It could be good to open an issue about this, unlikely that it breaks syncing but its possible

Reply:
The context for this comment seems to have been lost as I don't see any changes about lCost here now, but the comment isn't marked as outdated.  We can't merge if there's any risk of breaking syncing though. @tynes if this has been resolved can you resolve the conversation or otherwise can you give me some more detail on the issue here please?

Reply:
@tynes I also don't see the context. Has this been addressed, or what's the issue in more detail? @teddyknox maybe you know?